scripts/build

```diff
@@ -67,492 +67,489 @@ if [ -f $STAMP ] ; then
-            ;;
-
-          # make based builds
-          "configure:target"|"cmake-make:target"|"autotools:target"|"make:target")
-            $MAKEINSTALL $PKG_MAKEINSTALL_OPTS_TARGET
-            make install DESTDIR=$INSTALL $PKG_MAKEINSTALL_OPTS_TARGET
-            ;;
-          "configure:host"|"cmake-make:host"|"autotools:host"|"make:host")
-            make install $PKG_MAKEINSTALL_OPTS_HOST
-            ;;
-          "configure:init"|"cmake-make:init"|"autotools:init"|"make:init")
-            make install DESTDIR=$INSTALL $PKG_MAKEINSTALL_OPTS_INIT
-            ;;
-          "configure:bootstrap"|"cmake-make:bootstrap"|"autotools:bootstrap"|"make:bootstrap")
-            make install $PKG_MAKEINSTALL_OPTS_BOOTSTRAP
-            ;;
-        esac
-      fi
-      if [ "$(type -t post_makeinstall_$TARGET)" = "function" ]; then
-        post_makeinstall_$TARGET
-      fi
-
-      if [ "$TARGET" = "target" -o "$TARGET" = "init" ]; then
-        if [ -d $INSTALL ] ; then
-          rm -rf $INSTALL/{usr/,}include
-          rm -rf $INSTALL/{usr/,}lib/cmake
-          rm -rf $INSTALL/{usr/,}lib/pkgconfig
-          rm -rf $INSTALL/{usr/,}man
-          rm -rf $INSTALL/{usr/,}share/aclocal
-          rm -rf $INSTALL/{usr/,}share/bash-completion
-          rm -rf $INSTALL/{usr/,}share/doc
-          rm -rf $INSTALL/{usr/,}share/gtk-doc
-          rm -rf $INSTALL/{usr/,}share/info
-          rm -rf $INSTALL/{usr/,}share/locale
-          rm -rf $INSTALL/{usr/,}share/man
-          rm -rf $INSTALL/{usr/,}share/pkgconfig
-          rm -rf $INSTALL/{usr/,}share/zsh
-          rm -rf $INSTALL/{usr/,}var
-          find $INSTALL \( -name "*.orig" \
-                        -o -name "*.rej" \
-                        -o -name "*.a" \
-                        -o -name "*.la" \
-                        -o -name "*.o" \
-                        -o -name "*.in" \
-                        -o -name ".git*" \) \
-            -exec rm -f {} \; 2>/dev/null || :
-          find $INSTALL -type d -exec rmdir -p {} \; 2>/dev/null || :
-
-          if [ ! "$DEBUG" = yes ]; then
-            $STRIP `find $INSTALL \
-              -type f -name "*.so*" \
-              ! -name "ld-*.so" \
-              ! -name "libc-*.so" \
-              ! -name "libpthread-*.so" \
-              ! -name "libthread_db-*so" \
-              2>/dev/null` 2>/dev/null || :
-            if [ "$TARGET" = "init" ]; then
-              $STRIP `find $INSTALL -type f -name "*.so*" 2>/dev/null` 2>/dev/null || :
-            fi
-            $STRIP `find $INSTALL/bin $INSTALL/usr/bin $INSTALL/sbin $INSTALL/usr/sbin \
-              -type f -executable 2>/dev/null` 2>/dev/null || :
-          fi
-        fi
-      fi
+# build dependencies
+case "$TARGET" in
+  "target") _pkg_depends="$PKG_DEPENDS_TARGET";;
+  "host") _pkg_depends="$PKG_DEPENDS_HOST";;
+  "init") _pkg_depends="$PKG_DEPENDS_INIT";;
+  "bootstrap") _pkg_depends="$PKG_DEPENDS_BOOTSTRAP";;
```

Comment:
I'd prefer a catch-all here:
```
  *) _pkg_depends=;;
```
or initialise `_pkg_depends=` (or `unset _pkg_depends`) before the `case`. It shouldn't be set to anything, of course, but who knows what variables will be set when sourcing other files - the `package.mk` is sourced shortly before this section and it can set anything it likes.

I'd even be inclined to wrap the `for` loop in a test, just for clarity:
```
if [ -n "$_pkg_depends" ]; then
  <for loop>
fi
```
bash will drop straight through the loop when `_pkg_depends` is blank or unset, but it avoids any confusion...

Reply:
I am not sure about the the extra if. I do not like extra code.
The unset is added, will commit later.